source/csharp/Camera/Basic/CaptureViaGenICam/CaptureViaGenICam.cs

```diff
@@ -0,0 +1,128 @@
+            HalconDotNet.HOperatorSet.SetFramegrabberParam(framegrabber, "add_objectmodel3d_overlay_attrib", "enable");
+            HalconDotNet.HOperatorSet.SetFramegrabberParam(framegrabber, "AcquisitionMode", "SingleFrame");
+
+            Console.WriteLine("Configuring camera settings");
+            HalconDotNet.HOperatorSet.SetFramegrabberParam(framegrabber, "Aperture", 5.66);
+            HalconDotNet.HOperatorSet.SetFramegrabberParam(framegrabber, "ExposureTime", 8333);
+            HalconDotNet.HOperatorSet.SetFramegrabberParam(framegrabber, "Gain", 1);
+            HalconDotNet.HOperatorSet.SetFramegrabberParam(framegrabber, "ProcessingFiltersOutlierRemovalEnabled", 1);
+            HalconDotNet.HOperatorSet.SetFramegrabberParam(framegrabber, "ProcessingFiltersOutlierRemovalThreshold", 5);
+
+            Console.WriteLine("Capturing frame");
+            var region = new HalconDotNet.HObject();
+            var contours = new HalconDotNet.HObject();
+            var data = new HalconDotNet.HTuple();
+            var frame = new HalconDotNet.HObject();
+            HalconDotNet.HOperatorSet.GrabData(out frame, out region, out contours, framegrabber, out data);
+
+            var x = frame.SelectObj(1);
+            var y = frame.SelectObj(2);
+            var z = frame.SelectObj(3);
+            var snr = frame.SelectObj(4);
+            var rgb = frame.SelectObj(5);
+
+            Console.WriteLine("Removing invalid 3D points (zeroes)");
+            var reducedRegion = new HalconDotNet.HObject();
+            var zReduced = new HalconDotNet.HObject();
+            HalconDotNet.HOperatorSet.Threshold(z, out reducedRegion, 0.0001, 10000);
+            HalconDotNet.HOperatorSet.ReduceDomain(z, reducedRegion, out zReduced);
+
+            Console.WriteLine("Constructing ObjetModel3D based on XYZ data");
+            var objectModel3D = new HalconDotNet.HTuple();
+            HalconDotNet.HOperatorSet.XyzToObjectModel3d(x, y, zReduced, out objectModel3D);
+
+            Console.WriteLine("Adding RGB to ObjectModel3D");
+            SetColorsInObjectModel3D(objectModel3D, rgb, zReduced);
+
+            string fileName = "Zivid3D.ply";
+            Console.WriteLine("Saving point cloud to: " + fileName);
+            SaveHalconPointCloud(objectModel3D, fileName);
+        }
+        catch (Exception ex)
+        {
+            Console.WriteLine("Error: " + ex.Message);
+            Environment.ExitCode = 1;
+        }
+    }
+
+    private static void SaveHalconPointCloud(HalconDotNet.HTuple model, string fileName)
+    {
+        HalconDotNet.HOperatorSet.WriteObjectModel3d(model, "ply", fileName, "invert_normals", "false");
+    }
+
+    private static void SetColorsInObjectModel3D(HalconDotNet.HTuple objectModel3D, HalconDotNet.HObject RGB, HalconDotNet.HObject zReduced)
+    {
+        var rows = new HalconDotNet.HTuple();
+        var cols = new HalconDotNet.HTuple();
+        var domain = new HalconDotNet.HObject();
```

Comment:
change the order so to be the same in input arguments in the function that is using it?

Reply:
done